### ISSUE TITLE
Prepare for submission to ropensci

### ISSUE THREAD

**Issue by kauedesousa** [ropensci/chirps#5]
The package is almost ready for submission to ropensci. This is a Milestone proposed to R. Bivand in his course on "Analysing Spatial Data". The package will be also submitted to JOSS. Here some things to do.

To do: 

* Write vignette (done!)
* Write paper
* Check consistency with ropensci templates
* Check text and typos

**Comment by adamhsparks** [comment]
Happy to help with this since I've written several that are rOpenSci packages and reviewed a few as well.

**Comment by adamhsparks** [comment]
I have done the following so far today to help prepare for submission to rOpenSci:

* Add [codemeta.json](https://github.com/agrobioinfoservices/chirps/blob/master/codemeta.json) using {codemetar}

* [Precompute the vignette](https://github.com/agrobioinfoservices/chirps/blob/master/vignettes/precompile.R) so that installation is faster, see https://ropensci.org/technotes/2019/12/08/precompute-vignettes/

* Tidy up the vignette text

* Checked and corrected spelling using `devtools::spell_check()`

* Corrected minor issues in DESCRIPTION file such as grammar, author roles and line breaks at 80 chars

* Formatted documentation

**Comment by kauedesousa** [comment]
Great! I will start to write the paper.

**Comment by kauedesousa** [comment]
I just committed the first draft of the paper https://github.com/agrobioinfoservices/chirps/commit/e39feee5809782a149eb9ef1f1b39f406bd1f3a4

Tomorrow I will work on how to add the Fig1.png (or .svg) to the document.

**Comment by kauedesousa** [comment]
Added the fig but still some work to do.

**Comment by adamhsparks** [comment]
I've made a few very minor edits to paper.Rmd

**Comment by kauedesousa** [comment]
@adamhsparks, do you think we are ready to send it to ropensci?

**Comment by adamhsparks** [comment]
I've just pushed a last few minor changes and corrections for the paper and the tests.

Let's do it!

**Comment by kauedesousa** [comment]
Great! I will submit.

**Comment by kauedesousa** [comment]
Submitted https://github.com/ropensci/software-review/issues/357

**Comment by kauedesousa** [comment]
I have started the updates from the first comments/suggestions we got in the rOpenSci review. There is a folder [here](https://github.com/agrobioinfoservices/chirps/tree/master/reviewer_comments) with the point-to-point response to each comment. I couldn't go further because the server is not working. It generates the ids but do not return the data from the request (in R and using the example provided in the API documentation).

**Comment by adamhsparks** [comment]
Thanks.

I read through it. A few thoughts to help you.

1. The /man structure is part of using the pre-built vignettes. I'm not aware of any issues with it. CRAN hasn't baulked at it and I just submitted two package updates that use this method in the last month.

2. The tests use `skip_on_cran()` due to CRAN policies for Internet connectivity. You'll get a nastygram and have the package pulled if the test fails due to this so I just skip them on CRAN. Ideally webmocking would help with this. I tried to use and eventually failed/gave up with nasapower to use VCR, https://docs.ropensci.org/vcr/. But still I'd encourage us to look at implementing this in the tests. I'll try to have another go at it with chirps and see if I can get it to work properly.
